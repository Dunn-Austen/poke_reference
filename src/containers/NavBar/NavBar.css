header {
  align-items: center;
  background-color: gold;
  border-bottom: 2px inset black;
  box-sizing: border-box;
  display: flex;
  height: 7vh;
  justify-content: center;
  width: 100%;
}

.pokemon-name {
  text-transform: capitalize;
}

span {
  text-transform: lowercase;
}
