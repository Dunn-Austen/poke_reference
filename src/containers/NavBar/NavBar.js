import React, { Component } from 'react';
import './NavBar.css';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Route, Switch, Link } from 'react-router-dom';


export const NavBar = props => {
  const { pokeData } = props;

  return (
    <header>
      <Switch>
        <Route exact path='/' render={() =>
          <h1 className='welcome-text'>PokeReference</h1>
        }/>
        <Route exact path='/pokepage' render={() =>
          <div className='nav-container'>
            <Link to ='/'>
              <button className='home'>Return Home</button>
            </Link>
            <h1 className='pokemon-name'>{pokeData.name}</h1>
            <Link to ='/'>
              <button className='home'>Return Home</button>
            </Link>
          </div>
        }/>
        <Route component={() =>
          <h1 className='unefined-route'>Page Not Found</h1>
        }/>
      </Switch>
    </header>
  )
};

export const mapStateToProps = state => ({
  pokeData: state.pokeData
});

export default connect(mapStateToProps, null)(NavBar);

NavBar.propTypes = {
  pokeData: PropTypes.object
}
