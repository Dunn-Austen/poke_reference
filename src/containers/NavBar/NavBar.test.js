import React from 'react';
import { shallow } from 'enzyme';
import { NavBar, mapStateToProps } from './NavBar';

describe('NavBar', () => {

  describe('mapsStateToProps', () => {
    it('should return only the pertinent information from the redux store', () => {
      const mockState = {
        pokeNames: [{pokemon: 'dratini', url: 'url'}],
        errorMessage: '',
        isLoading: false,
        pokeTypes: [],
        pokeData: {},
        opponentTypes: [],
      };
      const expected = {
        pokeData: {},
      };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

})
