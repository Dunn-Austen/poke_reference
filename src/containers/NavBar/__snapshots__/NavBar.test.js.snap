// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match the NavBar Snapshot 1`] = `
<header>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/pokesearch"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/pokepage"
      render={[Function]}
    />
    <Route
      component={[Function]}
    />
  </Switch>
</header>
`;
